src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests2.cs

```diff
@@ -32432,5 +32432,493 @@ .method family hidebysig static void M () cil managed
         var method = comp.GlobalNamespace.GetTypeMember("E").GetTypeMembers().Single().GetMember<MethodSymbol>("M").GetPublicSymbol();
         Assert.Null(method.AssociatedExtensionImplementation);
     }
+
+    [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/78189")]
+    public void CallerArgumentExpression_01(bool useCompilationReference)
+    {
+        var callerSrc = """
+"A".Print();
+E.Print("B");
+"C".Print2();
+
+""";
+        var src = callerSrc + """
+public static class E
+{
+    public static void Print2(this string s, [System.Runtime.CompilerServices.CallerArgumentExpression(nameof(s))] string caller = "")
+    {
+        System.Console.Write($"{caller}={s} ");
+    }
+
+    extension(string s)
+    {
+        public void Print([System.Runtime.CompilerServices.CallerArgumentExpression(nameof(s))] string caller = "")
```

Comment:
Was other attributes (e.g, nullability attributes) considered/tested as well?

For example:

```
    extension(string s)
    {
        // If the state of the extension receiver is not null, then the return value nullability state should be not null as well.
        [return: NotNullIfNotNull(nameof(s))]
        public string? ReturnMe() => s;
```